app/workers/delivery_request_worker.rb

```diff
@@ -23,23 +27,21 @@ def self.queue_for_priority(priority)
     reschedule_job if msg["error_class"] == "RatelimitExceededError"
   end
 
-  def perform(email_id)
+  def perform(email_id, queue)
     @email = Email.find(email_id)
+    @queue = queue
     check_rate_limit!
     increment_rate_limiter
     DeliveryRequestService.call(email: @email)
   end
 
-  def self.perform_async_with_priority(*args, priority:)
-    @priority = priority
-    set(queue: queue_for_priority(priority))
-      .perform_async(*args)
+  def self.perform_async_for_immediate(*args, priority: :normal)
```

Comment:
Do we want to have this worker knowing about immediate/digest etc? I wonder if it would be better to have a more generic `perform_async_in_queue` (like Whitehall 🎉 ) and then take this logic out of here.

Reply:
I think I like these convenience methods as it helps make the existence of the queues more explicit. Particularly if we're having to manually send out emails on the console, these methods might make things easier to understand.

Reply:
I get the explicit bit but it does make this worker need to know about the types of emails we're handling and what their priorities should be rather than just delivering an email. We could make it explicit elsewhere.

Reply:
I've pushed a few commits now which gets rid of those specific methods, I agree it makes sense for the `DeliveryRequestWorker` to not know about digest and immediate emails.